sdk/extensions/Azure.Extensions.AspNetCore.DataProtection.Keys/tests/TypeForwardingActivatorTests.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Azure.Identity;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.DataProtection.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace Azure.Extensions.AspNetCore.DataProtection.Keys.Tests
+{
+    public class TypeForwardingActivatorTests : MarshalByRefObject
+    {
+        [Test]
+        public void CreateInstance_ForwardsToNewNamespaceIfExists()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddDataProtection().ProtectKeysWithAzureKeyVault(new Uri("http://localhost"), new DefaultAzureCredential());
+            var services = serviceCollection.BuildServiceProvider();
+            var activator = services.GetRequiredService<IActivator>();
+
+            // Act
+            var name = "Microsoft.AspNetCore.DataProtection.AzureKeyVault.AzureKeyVaultXmlDecryptor, Microsoft.AspNetCore.DataProtection.AzureKeyVault, Version=1.0.0.0";
+            var instance = activator.CreateInstance(typeof(object), name);
+
+            // Assert
+            Assert.IsInstanceOf<Azure.Extensions.AspNetCore.DataProtection.Keys.AzureKeyVaultXmlDecryptor>(instance);
+        }
+
+        [Test]
+        public void CreateInstance_DoesNotForwardIfClassDoesNotExist()
+        {
+            // Arrange
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddDataProtection();
+            var services = serviceCollection.BuildServiceProvider();
+            var activator = services.GetRequiredService<IActivator>();
+
+            // Act & Assert
+            var name = "Microsoft.AspNet.DataProtection.TypeForwardingActivatorTests+NonExistentClassWithParameterlessCtor, Microsoft.AspNet.DataProtection.Tests";
+            var exception = Assert.Throws(Is.InstanceOf<Exception>(), () => activator.CreateInstance(typeof(object), name));
```

Comment:
You could get more specific to make sure we keep throwing the same exception type if people are catching it.